src/vs/workbench/browser/codeeditor.ts

```diff
@@ -212,6 +216,44 @@ export class FloatingClickWidget extends Widget implements IOverlayWidget {
 	}
 }
 
+export class FloatingClickMenu extends Disposable implements IEditorContribution {
+
+	static readonly ID = 'editor.contrib.floatingClickMenu';
+
+	constructor(
+		editor: ICodeEditor,
+		@IInstantiationService instantiationService: IInstantiationService,
+		@IMenuService menuService: IMenuService,
+		@IContextKeyService contextKeyService: IContextKeyService
+	) {
+		super();
+
+		const menu = menuService.createMenu(MenuId.EditorContent, contextKeyService);
+		const menuDisposables = new DisposableStore();
+		const renderMenuAsFloatingClickBtn = () => {
+			menuDisposables.clear();
+			if (!editor.hasModel() || editor.getOption(EditorOption.inDiffEditor)) {
+				return;
+			}
+			const actions: IAction[] = [];
+			createAndFillInActionBarActions(menu, { renderShortTitle: true, shouldForwardArgs: true }, actions);
+			if (actions.length === 0) {
+				return;
+			}
+			// todo@jrieken find a way to handle N actions, like showing a context menu
```

Comment:
Maybe use the kind of button-with-dropdown that the SCM view has.

Reply:
yeah, something along those lines. But not for now, for later